### PR TITLE
Split FilePathFactory into sst part and file store part

### LINKED ISSUE #8: FilePathFactory can be separate to SstFileFactory and MetaFileFactory
SstFileFactory is dynamically generated by partition and bucket.
But MetaFileFactory is static class for one FileStore.

### PR DESCRIPTION
**Pull request by tsreaper**
## What is the purpose of the change

This PR fixes #8 . Currently `FilePathFactory` contains method both for creating sst file paths and manifest file paths. However for sst files they only need methods about sst files and for file store it only need methods about manifest files.

## Brief change log

 - Split FilePathFactory into sst part and file store part

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

### CODE REVIEW

File: dynamic-core/src/main/java/com/alibaba/flink/dynamic/fliestore/utils/FileStorePathFactory.java [hunk 1]
```diff
@@ -19,17 +19,16 @@
...
-
-    Path newSstFile();
+public interface FileStorePathFactory {
 
     Path newManifestFile();
 
     Path newManifestList();
 
-    Path toSstFilePath(String sstFileName);
+    Path toSstFilePath(BinaryRowData partition, int bucket, String sstFileName);
```
Comment (JingsongLi) [inline dynamic-core/src/main/java/com/alibaba/flink/dynamic/fliestore/utils/FileStorePathFactory.java:31]: We can move this to `SstPathFactory`. Merge tree only needs to read sst file.

File: dynamic-core/src/main/java/com/alibaba/flink/dynamic/fliestore/mergetree/sst/SstPathFactory.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.alibaba.flink.dynamic.fliestore.mergetree.sst;
+
+import org.apache.flink.core.fs.Path;
+
+/** Factory which produces new {@link Path}s for sst files. */
+public interface SstPathFactory {
```
Comment (JingsongLi) [inline dynamic-core/src/main/java/com/alibaba/flink/dynamic/fliestore/mergetree/sst/SstPathFactory.java:24]: We can have implement classes for `SstPathFactory` and `FileStorePathFactory`.

File: dynamic-core/src/main/java/com/alibaba/flink/dynamic/fliestore/utils/FileStorePathFactory.java (hunk 1 above)
Comment (JingsongLi) [reply dynamic-core/src/main/java/com/alibaba/flink/dynamic/fliestore/utils/FileStorePathFactory.java:31]: `FileStorePathFactory` can provides a method: `SstPathFactory  createSstPathFactory(part, bucket)`

**Comment by JingsongLi** [comment]
We should remove `TestFilePathFactory` too, `Files.createTempDirectory` is bad in tests, It is easy to leave files behind

File: dynamic-core/src/main/java/com/alibaba/flink/dynamic/fliestore/mergetree/sst/SstPathFactory.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+/** Factory which produces new {@link Path}s for sst files. */
+public class SstPathFactory {
+
+    private final Path bucketDir;
+    private final UUID uuid;
+
+    private int pathCount;
+
+    public SstPathFactory(Path root, BinaryRowData partition, int bucket) {
```
Comment (JingsongLi) [inline dynamic-core/src/main/java/com/alibaba/flink/dynamic/fliestore/mergetree/sst/SstPathFactory.java:36]: we need a readable partition path, can we just pass `String partition` to here?

File: dynamic-core/src/main/java/com/alibaba/flink/dynamic/fliestore/mergetree/sst/SstPathFactory.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        this.bucketDir = new Path(root + "/part" + partition.hashCode() + "/bucket" + bucket);
```
Comment (JingsongLi) [inline dynamic-core/src/main/java/com/alibaba/flink/dynamic/fliestore/mergetree/sst/SstPathFactory.java:37]: `bucket-`

File: dynamic-core/src/main/java/com/alibaba/flink/dynamic/fliestore/mergetree/sst/SstPathFactory.java [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+/** Factory which produces new {@link Path}s for sst files. */
+public class SstPathFactory {
+
+    private final Path bucketDir;
+    private final UUID uuid;
+
+    private int pathCount;
+
+    public SstPathFactory(Path root, String partition, int bucket) {
+        this.bucketDir = new Path(root + "/part-" + partition + "/bucket-" + bucket);
```
Comment (JingsongLi) [inline dynamic-core/src/main/java/com/alibaba/flink/dynamic/fliestore/mergetree/sst/SstPathFactory.java:36]: remove "part-"

Comment (JingsongLi) [reply dynamic-core/src/main/java/com/alibaba/flink/dynamic/fliestore/mergetree/sst/SstPathFactory.java:36]: partition can be nullable

Comment (tsreaper) [reply dynamic-core/src/main/java/com/alibaba/flink/dynamic/fliestore/mergetree/sst/SstPathFactory.java:36]: It is ok for partition to be nullable. Java will handle this for string concatenation.

Comment (JingsongLi) [reply dynamic-core/src/main/java/com/alibaba/flink/dynamic/fliestore/mergetree/sst/SstPathFactory.java]: what we want is:
partition: "dt=2020" => /root/dt=2020/bucket-0/...
no partition: /root/bucket-0/...

File: dynamic-core/src/main/java/com/alibaba/flink/dynamic/fliestore/mergetree/sst/SstPathFactory.java [hunk 6]
```diff
@@ -32,8 +33,16 @@
...
+    }
+
     public SstPathFactory(Path root, String partition, int bucket) {
-        this.bucketDir = new Path(root + String.valueOf(partition) + "/bucket-" + bucket);
+        this(new Path(root + "/" + partition + "/bucket-" + bucket));
+    }
+
+    private SstPathFactory(Path bucketDir) {
+        this.bucketDir = bucketDir;
         this.uuid = UUID.randomUUID();
```
Comment (JingsongLi) [inline dynamic-core/src/main/java/com/alibaba/flink/dynamic/fliestore/mergetree/sst/SstPathFactory.java:46]: uuid can be a string.

**Pull request merged**